### ISSUE TITLE
Syntax Highlighting in Visual Studio Code

### ISSUE THREAD

**Issue by chgibb** [AssemblyScript/assemblyscript#95]
Telling Visual Studio Code to treat Assemblyscript files as Typescript mostly works.
The issue being recognition of Assemblyscript's types.
![selection_009](https://user-images.githubusercontent.com/18079458/28745615-c7ce86a0-7449-11e7-8d61-dfec3a65386a.png)

Adding a triple-slash reference to the Assemblyscript types fixes the syntax highlighting.
![selection_007](https://user-images.githubusercontent.com/18079458/28745596-60de9f52-7449-11e7-8629-46646b87a09c.png)

However trying to compile the above ```.as``` module using the same small compiler I mentioned in #94 results in the following when the triple-slash ref is present:
```
assembly.d.ts(11,14): error TS2300: Duplicate identifier 'sbyte'.
assembly.d.ts(13,14): error TS2300: Duplicate identifier 'byte'.
assembly.d.ts(15,14): error TS2300: Duplicate identifier 'short'.
assembly.d.ts(17,14): error TS2300: Duplicate identifier 'ushort'.
assembly.d.ts(19,14): error TS2300: Duplicate identifier 'int'.
assembly.d.ts(21,14): error TS2300: Duplicate identifier 'uint'.
assembly.d.ts(23,14): error TS2300: Duplicate identifier 'long'.
assembly.d.ts(25,14): error TS2300: Duplicate identifier 'ulong'.
assembly.d.ts(27,14): error TS2300: Duplicate identifier 'bool'.
assembly.d.ts(29,14): error TS2300: Duplicate identifier 'float'.
assembly.d.ts(31,14): error TS2300: Duplicate identifier 'double'.
assembly.d.ts(33,14): error TS2300: Duplicate identifier 'uintptr'.
assembly.d.ts(38,14): error TS2300: Duplicate identifier 'int8'.
assembly.d.ts(40,14): error TS2300: Duplicate identifier 'uint8'.
assembly.d.ts(42,14): error TS2300: Duplicate identifier 'int16'.
assembly.d.ts(44,14): error TS2300: Duplicate identifier 'uint16'.
assembly.d.ts(46,14): error TS2300: Duplicate identifier 'int32'.
assembly.d.ts(48,14): error TS2300: Duplicate identifier 'uint32'.
assembly.d.ts(50,14): error TS2300: Duplicate identifier 'int64'.
assembly.d.ts(52,14): error TS2300: Duplicate identifier 'uint64'.
assembly.d.ts(54,14): error TS2300: Duplicate identifier 'float32'.
assembly.d.ts(56,14): error TS2300: Duplicate identifier 'float64'.
assembly.d.ts(61,15): error TS2451: Cannot redeclare block-scoped variable 'NaN'.
assembly.d.ts(63,15): error TS2451: Cannot redeclare block-scoped variable 'NaNf'.
assembly.d.ts(65,15): error TS2451: Cannot redeclare block-scoped variable 'Infinity'.
assembly.d.ts(67,15): error TS2451: Cannot redeclare block-scoped variable 'Infinityf'.
assembly.d.ts(72,15): error TS2300: Duplicate identifier 'Array'.
assembly.d.ts(93,15): error TS2300: Duplicate identifier 'Int8Array'.
assembly.d.ts(95,15): error TS2300: Duplicate identifier 'Uint8Array'.
assembly.d.ts(97,15): error TS2300: Duplicate identifier 'Int16Array'.
assembly.d.ts(99,15): error TS2300: Duplicate identifier 'Uint16Array'.
assembly.d.ts(101,15): error TS2300: Duplicate identifier 'Int32Array'.
assembly.d.ts(103,15): error TS2300: Duplicate identifier 'Uint32Array'.
assembly.d.ts(105,15): error TS2300: Duplicate identifier 'Int64Array'.
assembly.d.ts(107,15): error TS2300: Duplicate identifier 'Uint64Array'.
assembly.d.ts(109,15): error TS2300: Duplicate identifier 'Float32Array'.
assembly.d.ts(111,15): error TS2300: Duplicate identifier 'Float64Array'.
assembly.d.ts(116,15): error TS2300: Duplicate identifier 'String'.
node_modules/assemblyscript/assembly.d.ts(11,14): error TS2300: Duplicate identifier 'sbyte'.
node_modules/assemblyscript/assembly.d.ts(13,14): error TS2300: Duplicate identifier 'byte'.
node_modules/assemblyscript/assembly.d.ts(15,14): error TS2300: Duplicate identifier 'short'.
node_modules/assemblyscript/assembly.d.ts(17,14): error TS2300: Duplicate identifier 'ushort'.
node_modules/assemblyscript/assembly.d.ts(19,14): error TS2300: Duplicate identifier 'int'.
node_modules/assemblyscript/assembly.d.ts(21,14): error TS2300: Duplicate identifier 'uint'.
node_modules/assemblyscript/assembly.d.ts(23,14): error TS2300: Duplicate identifier 'long'.
node_modules/assemblyscript/assembly.d.ts(25,14): error TS2300: Duplicate identifier 'ulong'.
node_modules/assemblyscript/assembly.d.ts(27,14): error TS2300: Duplicate identifier 'bool'.
node_modules/assemblyscript/assembly.d.ts(29,14): error TS2300: Duplicate identifier 'float'.
node_modules/assemblyscript/assembly.d.ts(31,14): error TS2300: Duplicate identifier 'double'.
node_modules/assemblyscript/assembly.d.ts(33,14): error TS2300: Duplicate identifier 'uintptr'.
node_modules/assemblyscript/assembly.d.ts(38,14): error TS2300: Duplicate identifier 'int8'.
node_modules/assemblyscript/assembly.d.ts(40,14): error TS2300: Duplicate identifier 'uint8'.
node_modules/assemblyscript/assembly.d.ts(42,14): error TS2300: Duplicate identifier 'int16'.
node_modules/assemblyscript/assembly.d.ts(44,14): error TS2300: Duplicate identifier 'uint16'.
node_modules/assemblyscript/assembly.d.ts(46,14): error TS2300: Duplicate identifier 'int32'.
node_modules/assemblyscript/assembly.d.ts(48,14): error TS2300: Duplicate identifier 'uint32'.
node_modules/assemblyscript/assembly.d.ts(50,14): error TS2300: Duplicate identifier 'int64'.
node_modules/assemblyscript/assembly.d.ts(52,14): error TS2300: Duplicate identifier 'uint64'.
node_modules/assemblyscript/assembly.d.ts(54,14): error TS2300: Duplicate identifier 'float32'.
node_modules/assemblyscript/assembly.d.ts(56,14): error TS2300: Duplicate identifier 'float64'.
node_modules/assemblyscript/assembly.d.ts(61,15): error TS2451: Cannot redeclare block-scoped variable 'NaN'.
node_modules/assemblyscript/assembly.d.ts(63,15): error TS2451: Cannot redeclare block-scoped variable 'NaNf'.
node_modules/assemblyscript/assembly.d.ts(65,15): error TS2451: Cannot redeclare block-scoped variable 'Infinity'.
node_modules/assemblyscript/assembly.d.ts(67,15): error TS2451: Cannot redeclare block-scoped variable 'Infinityf'.
node_modules/assemblyscript/assembly.d.ts(72,15): error TS2300: Duplicate identifier 'Array'.
node_modules/assemblyscript/assembly.d.ts(93,15): error TS2300: Duplicate identifier 'Int8Array'.
node_modules/assemblyscript/assembly.d.ts(95,15): error TS2300: Duplicate identifier 'Uint8Array'.
node_modules/assemblyscript/assembly.d.ts(97,15): error TS2300: Duplicate identifier 'Int16Array'.
node_modules/assemblyscript/assembly.d.ts(99,15): error TS2300: Duplicate identifier 'Uint16Array'.
node_modules/assemblyscript/assembly.d.ts(101,15): error TS2300: Duplicate identifier 'Int32Array'.
node_modules/assemblyscript/assembly.d.ts(103,15): error TS2300: Duplicate identifier 'Uint32Array'.
node_modules/assemblyscript/assembly.d.ts(105,15): error TS2300: Duplicate identifier 'Int64Array'.
node_modules/assemblyscript/assembly.d.ts(107,15): error TS2300: Duplicate identifier 'Uint64Array'.
node_modules/assemblyscript/assembly.d.ts(109,15): error TS2300: Duplicate identifier 'Float32Array'.
node_modules/assemblyscript/assembly.d.ts(111,15): error TS2300: Duplicate identifier 'Float64Array'.
node_modules/assemblyscript/assembly.d.ts(116,15): error TS2300: Duplicate identifier 'String'.

/home/gibb/Desktop/asctesting/test2/node_modules/binaryen/index.js:26
function(e){if(!(e instanceof ha))throw e;});a.inspect=function(){return"[Emscripten Module object]"}}else if(La)a.print||(a.print=print),"undefined"!=typeof printErr&&(a.printErr=printErr),a.read="undefined"!=typeof read?read:function(){throw"no read() available";},a.readBinary=function(e){if("function"===typeof readbuffer)return new Uint8Array(readbuffer(e));e=read(e,"binary");v("object"===typeof e);return e},"undefined"!=typeof scriptArgs?a.arguments=scriptArgs:"undefined"!=typeof arguments&&(a.arguments=
                                  ^

Error: module.as compilation failed
    at Object.<anonymous> (/home/gibb/Desktop/asctesting/test2/index.js:15:11)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
```
The module compiles and runs fine without the triple-slash reference present. I assume this is due to the compiler defining the types somewhere itself, which causes the triple-slash to be a redefinition.

**Comment by dcodeIO** [comment]
That assumption is correct, the compiler includes assembly.d.ts on its own. If a `tsconfig.json` just works with `.as` files as well, this could solve the issue.

**Comment by chgibb** [comment]
I looked at using a ```tsconfig.json``` as in your documentation but I believe it broke ```.ts``` file compilation in the same project. My use case is to have ```.ts``` and ```.as``` coexisting in the same directory structure, just using different compilers to compile them.

**Comment by dcodeIO** [comment]
Does it work if you instead reference a path? I noticed that I did this and it seems to work.

```ts
/// <reference path="node_modules/assemblyscript/assembly" />
```

**Comment by chgibb** [comment]
Appears to be same errors as before.

**Comment by yahiko00** [comment]
I encountered this same issue when I started playing with AssemblyScript. But I think this is not really an issue from AssemblyScript itself.
1. It is not really correct to explicitly "redefine" builtin types by including a declaration file. For instance, the builtin type `float` has a specific meaning in AssemblyScript. It is not the same as the TypeScript type `number`.
2. Ideally, we should have an AssemblyScript equivalent of _tsserver.js_ to let Visual Studio Code highlighting correctly specific types. But I can guess this would be not trivial to implement...

**Comment by chgibb** [comment]
I fully agree with you @yahiko00 . Having the types present however makes it far easier to use Assemblyscript in unison with Typescript however, at least until something like an ```asserver``` appears which is non trivial, as you said.

For now I think having types defined is the best way to drive integration with Typescript.

**Comment by chgibb** [comment]
@dcodeIO I noticed your commit potentially addressing this. Do you want me to try it out on my setup?

**Comment by dcodeIO** [comment]
Sure. Haven't tested it, yet.

**Comment by chgibb** [comment]
Sweet. Will do. Is a version with that commit on NPM?

**Comment by chgibb** [comment]
Appears fixed in 0.3.0. Thanks!
![fixedhighlighting](https://user-images.githubusercontent.com/18079458/30344819-d4b7c4bc-97d0-11e7-9466-b4057e7f3aeb.png)
